nilearn/datasets/neurovault.py

```diff
@@ -969,15 +971,28 @@ def _get_batch(query, prefix_msg="", timeout=10.0, verbose=3):
     )
     prepped = session.prepare_request(req)
     logger.log(
-        f"{prefix_msg}getting new batch: {query}",
+        f"{prefix_msg}getting new batch:\n\t{query}",
         verbose=verbose,
         msg_level=_DEBUG,
-        stack_level=7,
+        stack_level=6,
     )
     try:
         resp = session.send(prepped, timeout=timeout)
         resp.raise_for_status()
         batch = resp.json()
+    except requests.exceptions.ReadTimeout:
+        logger.log(
+            (
+                f"Could not get batch from {query}.\n\t"
+                f"Timeout error with {timeout=} seconds.\n\t"
+                f"Try increasing 'timeout' value."
+            ),
+            msg_level=_ERROR,
+            verbose=verbose,
+            with_traceback=False,
```

Comment:
turning traceback off on this one to avoid error messages that are longer then my will to live

Reply:
I guess stack_level is not used then?

Reply:
(should be stacklevel I think so probably had no effect before anyway)

Reply:
stack_level also affect what prefix is printed in the message, so it will be used even with no traceback

Reply:
> that too could be cleaned up

you mean using the standard library logger?

Reply:
yeah. the advantage is that users then have control over which messages they want to see or not, how they should be formatted, which streams or files they should go to. it's more of one of those low-priority things we'll probably not get around too, though

Reply:
I think you already mentioned that in another issue actually

Reply:
I think we had discussed something similar in here https://github.com/nilearn/nilearn/pull/3702#pullrequestreview-1392745784

I would prefer to use the standard library logger in places where it does not affect how we interact with libraries that rely on using a "verbose" parameter.

> users then have control over which messages they want to see or not

For the neurovault API in our code base 'verbose' does this, so we could let verbose set the log level and rely on standard logging.